File: TransCoder_tokenized_test_set_functions/cpp/NUMBER_FULL_BINARY_TREES_NODE_PRODUCT_CHILDREN.cpp
int numoffbt ( int arr [ ], int n ) {
  int maxvalue = INT_MIN, minvalue = INT_MAX;
  for ( int i = 0;
  i < n;
  i ++ ) {
    maxvalue = max ( maxvalue, arr [ i ] );
    minvalue = min ( minvalue, arr [ i ] );
  }
  int mark [ maxvalue + 2 ];
  int value [ maxvalue + 2 ];
  memset ( mark, 0, sizeof ( mark ) );
  memset ( value, 0, sizeof ( value ) );
  for ( int i = 0;
  i < n;
  i ++ ) {
    mark [ arr [ i ] ] = 1;
    value [ arr [ i ] ] = 1;
  }
  int ans = 0;
  for ( int i = minvalue;
  i <= maxvalue;
  i ++ ) {
    if ( mark [ i ] ) {
      for ( int j = i + i;
      j <= maxvalue && j / i <= i;
      j += i ) {
        if ( ! mark [ j ] ) continue;
        value [ j ] = value [ j ] + ( value [ i ] * value [ j / i ] );
        if ( i != j / i ) value [ j ] = value [ j ] + ( value [ i ] * value [ j / i ] );
      }
    }
    ans += value [ i ];
  }
  return ans;
}

File: TransCoder_tokenized_test_set_functions/python/NUMBER_FULL_BINARY_TREES_NODE_PRODUCT_CHILDREN.py
def numoffbt ( arr, n ) :
    maxvalue = - 2147483647
    minvalue = 2147483647
    for i in range ( n ) :
        maxvalue = max ( maxvalue, arr [ i ] )
        minvalue = min ( minvalue, arr [ i ] )
    mark = [ 0 for i in range ( maxvalue + 2 ) ]
    value = [ 0 for i in range ( maxvalue + 2 ) ]
    for i in range ( n ) :
        mark [ arr [ i ] ] = 1
        value [ arr [ i ] ] = 1
    ans = 0
    for i in range ( minvalue, maxvalue + 1 ) :
        if ( mark [ i ] != 0 ) :
            j = i + i
            while ( j <= maxvalue and j //i<=i):
                if(mark[j]==0):
                    continue
                value[j]=value[j]+(value[i]*value[j//i])
                if(i!=j//i):
                    value[j]=value[j]+(value[i]*value[j//i])
                j+=i
        ans+=value[i]
    returnans
